quipucords/api/aggregate_report/model.py

```diff
@@ -0,0 +1,275 @@
+
+    os_by_name_and_version = defaultdict(lambda: defaultdict(int))
+    system_creation_dates = []
+    vmware_vms_by_host = defaultdict(list)
+
+    report = Report.objects.get(pk=report_id)
+    fingerprints = report.deployment_report.system_fingerprints.all()
+    for fingerprint in fingerprints:
+        # Yes, we want *all* OS name/version combos regardless of `is_rhel`.
+        # This means output may include non-RHEL distros (e.g. Ubuntu, Debian).
+        # Note: Ansible sources are fingerprinted to store the Ansible controller
+        # version (e.g. "4.4.4") in SystemFingerprint.os_version, and that can
+        # results strange in output here like {"unknown": {"4.4.4": 1}}.
+        # TODO Does Ansible controller's version really belong there?
+        # See also: FingerprintTaskRunner._process_ansible_fact.
+        os_by_name_and_version[fingerprint.os_name or "unknown"][
+            fingerprint.os_version or "unknown"
+        ] += 1
+
+        # TODO Is there a better way to determine the underlying source type?
+        # Is there always at least 1 fingerprint.sources? are they always the same type?
+        # Should I be doing something more clever like iterating through the list
+        # and checking for any presence of each source type?
+        # Could there be multiple different source types in the same fingerprint?
+        # If I made a bad assumption here, that might invalidate everything else below.
+        source_type = fingerprint.sources[0].get("source_type")
+
+        if (
+            source_type
+            in (DataSources.NETWORK, DataSources.SATELLITE, DataSources.VCENTER)
+            and not fingerprint.is_redhat
+        ):
+            # For all other fingerprint facts we aggregate below here, we only want
+            # to count them if we positively identified RHEL during fingerprinting.
+            continue
+
+        if not fingerprint.cpu_core_count and source_type != DataSources.VCENTER:
+            # Do not penalize vCenter sources because they do not collect core counts.
+            aggregated.missing_cpu_core_count += 1
+
+        if not fingerprint.cpu_socket_count:
+            aggregated.missing_cpu_socket_count += 1
+
+        if not fingerprint.name:
+            aggregated.missing_name += 1
+
+        if not fingerprint.system_creation_date:
+            aggregated.missing_system_creation_date += 1
+        else:
+            system_creation_dates.append(fingerprint.system_creation_date)
+
+        cpu_core_count = fingerprint.cpu_core_count or 0
+        jboss_presence = {
+            product.name: bool(product.presence == Product.PRESENT)
+            for product in fingerprint.products.all()
+            if product.name in jboss_product_names
+        }
+        # For now (forever?) we do not care to count FUSE or BRMS.
+        # If we don't care about them, then why bother collecting the data?
+        # TODO Remove *all* other API/model/etc. code related to FUSE and BRMS?
```

Comment:
This is a good PM question.

AFAIK, BRMS can be safely removed - we just never got the thumbs up for doing so. As for FUSE, I wish we could have a YES or NO answer for removing it. FUSE isn't EOL yet, but will soon be. Not sure if it will have extended support...